mysql/datadog_checks/mysql/queries.py

```diff
@@ -17,6 +17,12 @@
 FROM     information_schema.tables
 GROUP BY table_schema"""
 
+SQL_QUERY_TABLE_SIZE = """\
+SELECT   CONCAT(table_schema,'.',table_name) as table_name,
+         IFNULL(index_length/1024/1024,0) AS index_size_mb,
+         IFNULL(data_length/1024/1024,0) AS data_size_mb
+FROM     information_schema.tables"""
+
```

Comment:
this query returns a ton of tables, which might not be super interesting (depending on your use case, I guess). For example it returns information about a bunch of non-user created tables. Do you think that adding some extra configuration here might be worth it? 

e.g.: I am thinking you could have something, which allows users to exclude non-user created tables that could, when used, add a filter to this query like `where table_schema not in ('information_schema', 'mysql', 'performance_schema')`. The other thing that you could do is allow users to opt-in to getting metrics for the meta-schemas like `information_schema`, and exclude them by default.

Reply:
> @aymeric-ledizes I've been running some test queries using
> 
> ```
> SELECT table_schema, table_name, IFNULL(index_length/1024/1024,0) AS index_size_mb, IFNULL(data_length/1024/1024,0) AS data_size_mb FROM information_schema.tables where table_schema not in ('information_schema', 'mysql', 'performance_schema');
> ```
> 
> on a mysql v8.0 database, and ran into a few problems:
> 
>     1. It looks like this is a database view only, and not a system wide view so we would have to iterate over all of the schemas to collect user table information per existing schema
> 
>     2. While running this query as the datadog user (with the currently required configured permissions) I was not able to view any user created table information, and was only able to do so if I ran the query as a user with read permissions on the table.
> 
> 
> Do you know if the behavior I described in #2 is also an issue on mariadb? Perhaps the "fix" here is to make this feature only available to mariadb users? Otherwise, I don't think this feature will be very useful to mysql users.


In https://github.com/DataDog/integrations-core/pull/10674/commits/8874ccce0ca46e7f7ef119d1747df2b559d3c5d6 I suggest to separate systems tables metrics in `extra_table_metrics` to reduce the number of metrics collected.

For the 2 points:
1. We are doing a query on the specific `information_schema` schema so no iteration is needed
2. Yes I added a conf comment to precise it and this is common for mysql and mariadb: Information schema allows you to see information on the database objects the user have access to.

Reply:
I like the `extra_table_size_metrics` **option**. But 2 remarks on its implementation:
- naming: this option is about metrics for the "system tables", so to be more precise maybe name it `system_table_size_metrics`
- why is this option creating different **metric names** ? It's the same metric for me, just not for the same tables (so, different labels/tags on metrics).

Reply:
The name starting with `extra_` is to use similar names as other options. Tags are the same but yes the name of the metric produce should be the same and not starting with extra_ I'll update this point

Reply:
OK for me 👍

Reply:
Hi sorry for the delay on this, I have been blocked on testing this locally for the last few days. I am planning to take another look tomorrow. Thanks for working with me on it!

Reply:
I'm with @frivoire I think `extra_table_size_metrics` provides very little information about what this option actually means. I think `system_table_size_metrics` would make more sense here